course/flow.py

```diff
@@ -1612,7 +1612,9 @@ def add_buttons_to_form(form, fpctx, flow_session, permissions):
     return form
 
 
-def create_flow_page_visit(request, flow_session, page_data):
+@retry_transaction_decorator()
+def create_flow_page_visit(request, flow_session, page_data,
+        answer_data, is_submitted_answer):
```

Comment:
Thanks for the informative suggestions. The simplest method is to revert it back and add 2 lines,  what about my latest push.